app.py

```diff
@@ -70,3 +72,39 @@ async def create_product(product_payload: ProductPayload, db: AsyncSession = Dep
         await session.commit()
         await session.refresh(db_product)
         return db_product
+
+
+@app.get("/products/{product_id}", response_model=Product)
+async def get_product(product_id: int, db: AsyncSession = Depends(get_db)):
+    product_model = await queries.search_product(db, product_id)
+
+    if not product_model:
+        return Response(status_code=404)
+
+    return Product.from_orm(product_model)
+
+
+@app.put("/products/{product_id}", response_model=Product)
+async def update_product(
+        product_id: int,
+        product_update: ProductUpdate,
```

Comment:
this model will not fit for PUT, because PUT requires all Product parameters to be passed